core/state_processor_rip7560.go

```diff
@@ -165,22 +166,43 @@ func BuyGasRip7560Transaction(chainConfig *params.ChainConfig, gp *GasPool, head
 }
 
 func ApplyRip7560ValidationPhases(chainConfig *params.ChainConfig, bc ChainContext, author *common.Address, gp *GasPool, statedb *state.StateDB, header *types.Header, tx *types.Transaction, cfg vm.Config) (*ValidationPhaseResult, error) {
-	/*** Nonce Validation Frame ***/
-	nonceValidationMsg := prepareNonceValidationMessage(tx, chainConfig)
 	blockContext := NewEVMBlockContext(header, bc, author, chainConfig, statedb)
-	txContext := NewEVMTxContext(nonceValidationMsg)
-	txContext.Origin = *tx.Rip7560TransactionData().Sender
+	txContext := vm.TxContext{
+		Origin:   *tx.Rip7560TransactionData().Sender,
+		GasPrice: tx.GasFeeCap(),
+	}
 	evm := vm.NewEVM(blockContext, txContext, statedb, chainConfig, cfg)
 
+	/*** Nonce Validation Frame ***/
+	nonceValidationMsg, useLegacyNonce := prepareNonceValidationMessage(tx, chainConfig)
 	var nonceValidationUsedGas uint64
-	resultNonceManager, err := ApplyMessage(evm, nonceValidationMsg, gp)
-	if err != nil {
-		return nil, err
-	}
-	if resultNonceManager.Err != nil {
-		return nil, resultNonceManager.Err
+	if !useLegacyNonce {
+		resultNonceManager, err := ApplyMessage(evm, nonceValidationMsg, gp)
+		if err != nil {
+			return nil, err
+		}
+		if resultNonceManager.Err != nil {
+			return nil, resultNonceManager.Err
+		}
+		nonceValidationUsedGas = resultNonceManager.UsedGas
+	} else {
```

Comment:
Why not declare something like useDefaultNonce on the msg object and let it ride the default logic on state_transition?

Reply:
Yeah i've thought about this, and should inspect more that which one will be better for less complexity. Maybe we can push it back to the next iteration.